nixos/lib/make-ext4-fs.nix

```diff
@@ -78,6 +78,15 @@ pkgs.stdenv.mkDerivation {
       # get rid of the unnecessary slack here--but see
       # https://github.com/NixOS/nixpkgs/issues/125121 for caveats.
 
+      # shrink to fit
+      resize2fs -M $img
+
+      # Add 16 MebiByte to the current_size
+      new_size=$(dumpe2fs -h $img | awk -F: \
+        '/Block count/{count=$2} /Block size/{size=$2} END{print (count*size+16*2**20)/size}')
+
+      resize2fs $img $new_size
+
```

Comment:
Why is this suddenly needed?

Reply:
It's not suddenly needed. However, this is what I think is required to 

> minimizing the size of the filesystem (while explicitly reserving some slack

Mentioned over here: https://github.com/NixOS/nixpkgs/pull/125121#issuecomment-855819167

Reply:
@samueldr ping?